plugin/arcconf/arcconf.py

```diff
@@ -375,18 +375,17 @@ def systems(self, flags=0):
         return rc_lsm_syss
 
     @staticmethod
-    def _arcconf_array_to_lsm_pool(ld_name, sys_id, ctrl_num):
-        pool_id = '%s:%s' % (ctrl_num, ld_name)
-        name = 'Array' + str(ld_name)
+    def _arcconf_array_to_lsm_pool(array_id, array_name, sys_id, block_size, total_size, unused_size):
+        pool_id = '%s:%s' % (sys_id, array_id)
+        name = 'Array ' + str(array_name)
         elem_type = Pool.ELEMENT_TYPE_VOLUME | Pool.ELEMENT_TYPE_VOLUME_FULL
         unsupported_actions = 0
-        free_space = 0
-        total_space = 0
-        # TODO Need to have a logic for finding free space and total space.
+        free_space = int(block_size) * int(unused_size)
+        total_space = int(block_size) * int(total_size)
 
         status = Pool.STATUS_OK
         status_info = ''
-        plugin_data = "%s:%s" % (ctrl_num, ld_name)
+        plugin_data = {'ctrl_id': sys_id, 'array_id': array_id}
```

Comment:
The `plugin_data` should be a `string` rather than a `dict`. Just use the `pool_id` would be enough.

Reply:
Done - commit id - e5c4436